habitat-lab/habitat/core/embodied_task.py

```diff
@@ -234,38 +235,31 @@ def __init__(
 
         self.measurements = Measurements(
             self._init_entities(
-                entity_names=config.measurements,
+                entities_configs=config.measurements,
                 register_func=registry.get_measure,
-                entities_config=config,
             ).values()
         )
 
         self.sensor_suite = SensorSuite(
             self._init_entities(
-                entity_names=config.sensors,
+                entities_configs=config.state_sensors,
                 register_func=registry.get_sensor,
-                entities_config=config,
             ).values()
         )
 
         self.actions = self._init_entities(
-            entity_names=config.possible_actions,
+            entities_configs=config.actions,
             register_func=registry.get_task_action,
-            entities_config=self._config.actions,
         )
         self._action_keys = list(self.actions.keys())
 
         self._is_episode_active = False
 
-    def _init_entities(
-        self, entity_names, register_func, entities_config=None
-    ) -> OrderedDict:
-        if entities_config is None:
-            entities_config = self._config
+    def _init_entities(self, entities_configs, register_func) -> OrderedDict:
 
         entities = OrderedDict()
-        for entity_name in entity_names:
-            entity_cfg = getattr(entities_config, entity_name)
+        for entity_name, entity_cfg in entities_configs.items():
```

Comment:
Big change that breaks all existing yaml (except for the pick.yaml that was converted). Instead of a dict + list of enabled, now we only have a dict.